sdk/lib/core/map.dart

```diff
@@ -261,15 +318,35 @@ abstract class Map<K, V> {
   /// and adds the key with the returned value to the map.
   ///
   /// If the key is not present, [ifAbsent] must be provided.
+  /// ```dart
+  /// final data = {1: 'A', 2: 'B', 3: 'C'};
+  /// data.update(2, (value) => 'ABCD', ifAbsent: () => 'G');
+  /// print(data); // {1: A, 2: ABCD, 3: C}
+  /// // if there is no key then ifAbsent is used
+  /// data.update(10, (value) => 'XYZ', ifAbsent: () => 'X');
+  /// print(data); // {1: A, 2: ABCD, 3: C, 10: X}
+  /// ```
   V update(K key, V update(V value), {V ifAbsent()?});
```

Comment:
This example needs work, and I think the text does too - me and Alex both weren't clear on what this method does.

- The text doesn't explain what `ifAbsent` actually does
- What happens if they key doesn't exist and you don't provide `ifAbsent`?
- It also doesn't explain what `value` is - I'm assuming it's the current value of that key? There could be an example of using the current value to update the map